src/extensions/displayExtensionInfo.ts

```diff
@@ -54,33 +109,59 @@ export async function displayExtInfo(
   if (roles.length) {
     lines.push(await displayRoles(roles));
   }
-  if (lines.length > 0) {
-    utils.logLabeledBullet(logPrefix, `information about '${clc.bold(extensionName)}':`);
-    const infoStr = lines.join("\n");
-    // Convert to markdown and convert any trailing newlines to a single newline.
-    const formatted = marked(infoStr).replace(/\n+$/, "\n");
-    logger.info(formatted);
-    // Return for testing purposes.
-    return lines;
-  } else {
-    throw new FirebaseError(
-      "Error occurred during installation: cannot parse info from source spec",
-      {
-        context: {
-          spec: spec,
-          extensionName: extensionName,
-        },
-      }
-    );
-  }
+  logger.info(`\n${lines.join("\n")}`);
+  return lines;
 }
 
-/**
- * Prints a clickable link where users can download the source code for an Extension Version.
- */
-export function printSourceDownloadLink(sourceDownloadUri: string): void {
-  const sourceDownloadMsg = `Want to review the source code that will be installed? Download it here: ${sourceDownloadUri}`;
-  utils.logBullet(marked(sourceDownloadMsg));
+export function displayExternalServices(spec: ExtensionSpec) {
+  const lines =
+    spec.externalServices?.map((service: ExternalService) => {
+      return `  - ${clc.cyan(`${service.name} (${service.pricingUri})`)}`;
+    }) ?? [];
+  return clc.bold("External services used:\n") + lines.join("\n");
+}
+
+export function displayEvents(spec: ExtensionSpec) {
+  const lines =
+    spec.events?.map((event: EventDescriptor) => {
+      return `  - ${clc.magenta(event.type)}${event.description ? `: ${event.description}` : ""}`;
+    }) ?? [];
+  return clc.bold("Events emitted:\n") + lines.join("\n");
+}
+
+export function displayResources(spec: ExtensionSpec) {
+  const lines = spec.resources.map((resource: Resource) => {
+    let type: string = resource.type;
+    switch (resource.type) {
+      case "firebaseextensions.v1beta.function":
+        type = "Cloud Function V1";
+        break;
+      case "firebaseextensions.v1beta.v2function":
+        type = "Cloud Function V2";
```

Comment:
I think the official product name is Cloud Function (1st gen)/(2nd gen)

ie: https://codelabs.developers.google.com/codelabs/cloud-starting-cloudfunctions-v2#0